apps/scratch/src/containers/customized-scratch-containers/Blocks.tsx

```diff
@@ -594,6 +592,25 @@ class Blocks extends React.Component<Props, State> {
     this.props.vm.runtime.off("PROJECT_LOADED", this.onProjectLoaded);
   }
 
+  attachWorkspaceListeners() {
+    const workspace = this.getWorkspace();
+
+    // Blockly does not provide a way to check if a listener is already attached.
+    // We remove and reattach to avoid duplicates without having to track attachment state
+    workspace.removeChangeListener(this.blockListener);
+    workspace.addChangeListener(this.blockListener);
```

Comment:
Do we need to do this? If I am not mistaken, before we just added listeners in some places and removed them in others?